### DISCUSSION TITLE
Benchmarking..

### DISCUSSION THREAD

**Discussion by JMarkin** [psqlpy-python/psqlpy#65] [Performance issues]
Hello, thank you for great job.

I added your driver to [pgbeanch](https://github.com/JMarkin/pgbench/blob/psqlpy/_python/pgbench_python.py#L212-L232) from MagicStack team. And result super different then your.
Full report https://public.jmarkin.ru/pgbeanch.html

Example result 
<img width="1019" alt="image" src="https://github.com/qaspen-python/psqlpy/assets/15740814/7cdfcd29-e840-404c-9da1-b34aab98927f">
psqlpy slower than all of other driver: aiopg, psycopg, asyncpg, etc.

Can you help me understand why your beanchmark give better results?

**Comment by chandr-andr** [comment]
@JMarkin Hello! Thanks for testing the library. I'm going to look into it.  
As for benchmark I made, I didn't want to show the raw performance of the library, so I tested it within FastAPI and not local PostgreSQL  

As far as I know, MagicStack benchmark is a little strange cuz it shows that psycopg 5! times slower than asyncpg, I haven't seen such outperformance in any test I've made.

**Comment by chandr-andr** [comment]
@JMarkin 
Plus, you can check this issue (https://github.com/piccolo-orm/piccolo/issues/986) with some discussion with the piccolo-orm creator and member, where we've tested asyncpg and psqlpy performance.  
The difference isn't big (I explained why) but it's here.

Last but not least, you can check this repo https://github.com/ymezencev/db_perf, it uses the old version of psqlpy, but it beats the asyncpg.

**Comment by chandr-andr** [comment]
Moved to discussions
